File: src/main/java/com/studyolle/modules/event/Enrollment.java
package com.studyolle.modules.event;

import com.studyolle.modules.account.Account;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.jpa.repository.EntityGraph;

import javax.persistence.*;
import java.time.LocalDateTime;

@NamedEntityGraph(
        name = "Enrollment.withEventAndStudy",
        attributeNodes = {
                @NamedAttributeNode(value = "event", subgraph = "study")
        },
        subgraphs = @NamedSubgraph(name = "study", attributeNodes = @NamedAttributeNode("study"))
)
@Entity
@Getter
@Setter
@EqualsAndHashCode(of = "id")
public class Enrollment {

    @Id @GeneratedValue
    private Long id;

    /* 테이블 enrollment : 컬럼 event_id */
    @ManyToOne
    private Event event;

    /* 테이블 enrollment : 컬럼 account_id */
    @ManyToOne
    private Account account;

    private LocalDateTime enrolledAt;

    private boolean accepted;

    private boolean attended;

}

File: src/main/kotlin/com/studyolle/domain/Enrollment.kt
package com.studyolle.domain

import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToOne

@Entity
class Enrollment(
    @Id
    @GeneratedValue
    val id: Long? = null,

    /* 테이블 enrollment : 컬럼 event_id */
    @ManyToOne
    val event: Event? = null,

    /* 테이블 enrollment : 컬럼 account_id */
    @ManyToOne
    val account: Account? = null,

    val enrolledAt: LocalDateTime? = null,
    val accepted : Boolean = false,
    val attended : Boolean = false,
) {
}